### PR TITLE
Header component

### PR DESCRIPTION
**Pull request by scottnash**

### CODE REVIEW

File: app/services/client/popup.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+class service {
+  getWindowSize() {
+    const dualScreenLeft = window.screenLeft !== undefined ? window.screenLeft : screen.left,
+      dualScreenTop = window.screenTop !== undefined ? window.screenTop : screen.top,
+      width = window.innerWidth || document.documentElement.clientWidth || screen.width,
+      height = window.innerHeight || document.documentElement.clientHeight || screen.height;
+
+    return {dualScreenLeft, dualScreenTop, height, width};
```
Comment (manuelurenah) [inline app/services/client/popup.js]: Please mind the spacing
```suggestion
    return { dualScreenLeft, dualScreenTop, height, width };
```

File: app/services/client/popup.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  getCenterPosition(windowSize, dimensions) {
+    const left = windowSize.width / 2 - dimensions.w / 2 + windowSize.dualScreenLeft,
+      top = windowSize.height / 2 - dimensions.h / 2 + windowSize.dualScreenTop;
+
+    return { left, top};
```
Comment (manuelurenah) [inline app/services/client/popup.js]:
```suggestion
    return { left, top };
```

File: app/services/client/sharePopup.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    this.setDimensions();
+    this.addShareURL();
+    this.addClickHandler();
+  }
+
+  addShareURL() {
+    switch (this.shareService) {
+      case 'twitter':
+        this.shareLink.href = 'https://twitter.com/share?text=' + encodeURIComponent(this.shareTitle) + '&url=' + this.shareURL + '?utm_source=tw&utm_medium=s3&utm_campaign=sharebutton-t';
```
Comment (manuelurenah) [inline app/services/client/sharePopup.js]: What do you think if we use string literal in here?
```suggestion
        this.shareLink.href = `https://twitter.com/share?text=${encodeURIComponent(this.shareTitle)}&url=${this.shareURL}?utm_source=tw&utm_medium=s3&utm_campaign=sharebutton-t`;
```

File: app/styleguides/_default/components/header.css [hunk 4]
```diff
@@ -1,77 +1,66 @@
...
-        display: block;
-        height: 43px;
-        margin: auto;
-        width: 43px;
-      }
+  svg {
+    height: auto;
+    width: 26px;
+  }
+  .share-link {
```
Comment (manuelurenah) [inline app/styleguides/_default/components/header.css]: Please add a new line above this one to separate css blocks

File: app/styleguides/_default/components/header.css [hunk 5]
```diff
@@ -1,77 +1,66 @@
...
+    margin-right: 20px;
+    position: relative;
+    span:last-child {
```
Comment (manuelurenah) [inline app/styleguides/_default/components/header.css]: Same here

File: app/components/header/model.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+module.exports.save = function (uri, data) {
+  if (data.siteLogo) {
+    data.siteLogo = data.siteLogo.trim();
+
+    if (data.siteLogo.substr(0,4).toLowerCase() === '<svg') {
+      data.isSVGString = true;
+    } else {
+      data.isSVGString = false;
+    }
```
[the remark this answers is not in the document]
Comment (scottnash) [reply app/components/header/model.js]: 👍

File: app/components/header/template.hbs [hunk 7]
```diff
@@ -1,9 +1,42 @@
...
+<header data-uri="{{ default _ref _self }}" class="{{componentVariation}} {{ pageType }}">
+  <div class="header-inner">
+    {{#if logoUrl}}
+    <a class="header__logo-link" href="{{logoUrl}}">
+    {{/if}}
+      <span class="header__title-logo">
+        {{#if isSVGString}}
+          {{{siteLogo}}}
+        {{else}}
+        <img class="heeader__logo" src="{{siteLogo}}">
```
[the remark this answers is not in the document]
Comment (scottnash) [reply app/components/header/template.hbs]: 👍

File: app/services/client/popup.js (hunk 1 above)
Comment (scottnash) [reply app/services/client/popup.js]: 👍

File: app/services/client/popup.js (hunk 2 above)
Comment (scottnash) [reply app/services/client/popup.js]: 👍

File: app/services/client/sharePopup.js (hunk 3 above)
Comment (scottnash) [reply app/services/client/sharePopup.js]: I think it's a good idea.

File: app/styleguides/_default/components/header.css (hunk 4 above)
Comment (scottnash) [reply app/styleguides/_default/components/header.css]: 👍

File: app/styleguides/_default/components/header.css (hunk 5 above)
Comment (scottnash) [reply app/styleguides/_default/components/header.css]: 👍

File: app/services/client/sharePopup.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+    this.setDimensions();
+    this.addShareURL();
+    this.addClickHandler();
+  }
+
+  addShareURL() {
+    switch (this.shareService) {
+      case 'twitter':
+        this.shareLink.href = `https://twitter.com/share?text=${encodeURIComponent(this.shareTitle)}&url='${this.shareURL}?utm_source=tw&utm_medium=s3&utm_campaign=sharebutton-t`;
```
Comment (manuelurenah) [inline app/services/client/sharePopup.js]: Oopsss, I think I left a single quote in there that's not necessary
```suggestion
        this.shareLink.href = `https://twitter.com/share?text=${encodeURIComponent(this.shareTitle)}&url=${this.shareURL}?utm_source=tw&utm_medium=s3&utm_campaign=sharebutton-t`;
```

File: app/services/client/popup.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    return { left, top };
+  }
+
+
+
+  /**
+ * openPopup Window
+ * @param {string} url - address of the popup page
+ * @param {object} dimensions { w: width of popup, h: height of popup}
+ */
```
Comment (manuelurenah) [inline app/services/client/popup.js]: I know that this is being picky, but can you please align the asterisks in here? 😛 
```js
/**
 * openPopup Window
 * @param {string} url - address of the popup page
 * @param {object} dimensions { w: width of popup, h: height of popup}
 */
```

**Pull request merged**